src/Security/Authorization/Policy/src/AuthorizationMiddlewareResultHandler.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authorization.Policy
+{
+    public class AuthorizationMiddlewareResultHandler : IAuthorizationMiddlewareResultHandler
+    {
+        public async Task HandleAsync(RequestDelegate requestDelegate, HttpContext httpContext, AuthorizationPolicy authorizationPolicy, PolicyAuthorizationResult policyAuthorizationResult)
+        {
+            if (policyAuthorizationResult.Succeeded)
```

Comment:
I would prefer if this code looked identical to what's currently in the middleware, does ExecuteHttpAction really need to be refactored out?

Reply:
used original code